trinity/protocol/eth/managers.py

```diff
@@ -57,38 +67,47 @@
 ]
 
 
-class GetBlockHeadersRequestManager(BaseGetBlockHeadersRequestManager):
+class HeaderRequestArgs(NamedTuple):
+    block_number_or_hash: BlockIdentifier
+    max_headers: int
+    skip: int
+    reverse: bool
+
+
+BlockHeadersRequestManager = RequestManager['ETHPeer', Tuple[BlockHeader, ...]]
+
+
+class GetBlockHeadersRequestManager(BaseService):
     msg_queue_maxsize = 100
 
     _response_msg_type: Type[Command] = BlockHeaders
 
-    # All `RequestManager` classes are expected to implement the `__call__`
-    # method, including changing the function signature, thus the
-    # `# type: ignore` here is both expected and required.
-    async def __call__(self,  # type: ignore
+    def __init__(self, peer: 'ETHPeer', token: CancelToken) -> None:
+        super().__init__(token)
+        self._peer = peer
+        self._request_manager = BlockHeadersRequestManager(peer, BlockHeaders, token)
+
+    async def __call__(self,
```

Comment:
So I think we can get rid of this class entirely by moving this functionality down into the `Handler` class, and rather than having `get_block_headers` be a class with a `__call__` we could just make it an actual method which handles this orchestration.  The `Handler` class would then become responsible for running the `_request_manager` which has the nice effect of removing this extra layer of services.

Reply:
The first roadblock I ran into is that the normalization step sometimes needs to run in a process pool, so it's easiest to call that from inside a service.